bundles/net.tourbook.cloud/src/net/tourbook/cloud/dropbox/PrefPageDropbox.java

```diff
@@ -15,55 +15,148 @@
  *******************************************************************************/
 package net.tourbook.cloud.dropbox;
 
+import com.sun.net.httpserver.HttpServer;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+
 import net.tourbook.cloud.Activator;
-import net.tourbook.cloud.IPreferences;
-import net.tourbook.cloud.oauth2.OAuth2BrowserDialog;
-import net.tourbook.cloud.oauth2.OAuth2Client;
+import net.tourbook.cloud.Preferences;
+import net.tourbook.cloud.oauth2.IOAuth2Constants;
+import net.tourbook.cloud.oauth2.OAuth2Utils;
+import net.tourbook.common.UI;
 import net.tourbook.common.util.StringUtils;
+import net.tourbook.web.WEB;
 
-import org.eclipse.jface.dialogs.MessageDialog;
 import org.eclipse.jface.layout.GridDataFactory;
 import org.eclipse.jface.layout.GridLayoutFactory;
-import org.eclipse.jface.preference.FieldEditorPreferencePage;
+import org.eclipse.jface.layout.PixelConverter;
 import org.eclipse.jface.preference.IPreferenceStore;
-import org.eclipse.jface.window.Window;
-import org.eclipse.osgi.util.NLS;
+import org.eclipse.jface.preference.PreferencePage;
+import org.eclipse.jface.util.IPropertyChangeListener;
+import org.eclipse.jface.util.PropertyChangeEvent;
 import org.eclipse.swt.SWT;
 import org.eclipse.swt.events.SelectionAdapter;
 import org.eclipse.swt.events.SelectionEvent;
 import org.eclipse.swt.widgets.Button;
 import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
 import org.eclipse.swt.widgets.Display;
-import org.eclipse.swt.widgets.Text;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
 import org.eclipse.ui.IWorkbench;
 import org.eclipse.ui.IWorkbenchPreferencePage;
 
-public class PrefPageDropbox extends FieldEditorPreferencePage implements IWorkbenchPreferencePage {
+public class PrefPageDropbox extends PreferencePage implements IWorkbenchPreferencePage {
+
+   public static final String      ID         = "net.tourbook.cloud.PrefPageDropbox";       //$NON-NLS-1$
 
-   public static final String ID         = "net.tourbook.cloud.PrefPageDropbox";       //$NON-NLS-1$
+   public static final String      ClientId   = "vye6ci8xzzsuiao";                          //$NON-NLS-1$
 
-   private IPreferenceStore   _prefStore = Activator.getDefault().getPreferenceStore();
+   private IPreferenceStore        _prefStore = Activator.getDefault().getPreferenceStore();
+   private IPropertyChangeListener _prefChangeListener;
+
+   private HttpServer              _server;
```

Comment:
Maybe but then we encounter the same case that you [commented here](https://github.com/wolfgang-ch/mytourbook/pull/277#discussion_r551010123) since Strava (any cloud vendor) also requires a static callback url

Reply:
I have implemented it for Strava as well. In this case, it has that big advantage **AND** the browser is less likely to have the same "deprecation" issue we saw with Dropbox :smile:

Reply:
I have updated the screenshots in the [PR](https://github.com/wolfgang-ch/mytourbook/pull/273) mentioned in the [21.next.0-readme.txt](https://github.com/wolfgang-ch/mytourbook/blob/f9cd17487c6142e177b70fe6d623f20b570e2c37/info/release-notes/21.next.0-readme.txt#L37)

Reply:
> I have implemented it for Strava as well. In this case, it has that big advantage **AND** the browser is less likely to have the same "deprecation" issue we saw with Dropbox 😄

It needed some work but now we are on the save side 👍